lib/libc/string/strlcat.c

```diff
@@ -32,23 +32,22 @@ __FBSDID("$FreeBSD$");
 size_t
 strlcat(char * __restrict dst, const char * __restrict src, size_t dsize)
 {
-	const char *odst = dst;
-	const char *osrc = src;
-	size_t n = dsize;
+	const char * const odst = dst;
+	const char * const osrc = src;
 	size_t dlen;
+	size_t n;
 
 	/* Find the end of dst and adjust bytes left but don't go past end. */
-	while (n-- != 0 && *dst != '\0')
+	for (n = dsize; n != 0 && *dst != '\0'; n--)
 		dst++;
 	dlen = dst - odst;
 	n = dsize - dlen;
 
-	if (n-- == 0)
+	if (n == 0)
 		return(dlen + strlen(src));
 	while (*src != '\0') {
-		if (n != 0) {
+		if (--n != 0) {
 			*dst++ = *src;
-			n--;
 		}
 		src++;
 	}
```

Comment:
Why? This has done nothing interesting to the code, and I'd expect it to compile to identical assembly.